src/test/java/org/broadinstitute/hellbender/utils/spark/SparkUtilsUnitTest.java

```diff
@@ -139,17 +141,31 @@ public void testPutReadsWithSameNameInSamePartition(int numPairs, int numPartiti
         return reads;
     }
 
-    @Test(expectedExceptions = GATKException.class)
-    public void testReadsPairsSpanningMultiplePartitionsCrash() {
+    @Test
+    public void testReadsPairsSpanningMultiplePartitions() {
         JavaSparkContext ctx = SparkContextFactory.getTestSparkContext();
         SAMFileHeader header = ArtificialReadUtils.createArtificialSamHeader();
         header.setSortOrder(SAMFileHeader.SortOrder.queryname);
         List<GATKRead> reads = createPairedReads(header, 40, 2);
-        // Creating one group in the middle that should cause problems
-        reads.addAll(40, createPairedReads(header, 1, 30));
+        reads.addAll(createPairedReads(header, 1, 30));
+        reads.sort( (r1, r2) -> r1.getName().compareTo(r2.getName()) );
 
         JavaRDD<GATKRead> problemReads = ctx.parallelize(reads,5 );
-        SparkUtils.putReadsWithTheSameNameInTheSamePartition(header, problemReads, ctx);
+        JavaRDD<GATKRead> fixedReads =
+                SparkUtils.putReadsWithTheSameNameInTheSamePartition(header, problemReads, ctx);
+        final List<Tuple2<String, Iterable<Integer>>> xPartitionPairs = fixedReads
+                .mapPartitionsWithIndex(( idx, itr ) -> {
+                    final Set<Tuple2<String, Integer>> readLocs = new HashSet<>();
+                    while ( itr.hasNext() ) readLocs.add(new Tuple2<>(itr.next().getName(), idx));
+                    return readLocs.iterator();
+                }, false)
+                .mapToPair(t -> t).groupByKey().filter(t -> {
+                    int count = 0;
+                    for ( final int idx : t._2() ) ++count;
+                    return count > 1;
+                })
+                .collect();
+        Assert.assertEquals(xPartitionPairs.size(), 0);
```

Comment:
It might be nice to also assert that each read name you expect is still there, since the fix you implemented involves removing some groups and re-adding them to different partitions.